torchdata/dataloader2/dataloader2.py

```diff
@@ -88,25 +121,16 @@ def __iter__(self) -> Iterator[T_co]:
                 self.reading_service.initialize_iteration()
 
             self._datapipe_iter = iter(self.datapipe)
-
             self._reset_iter = False
 
-        return self
-
-    def __next__(self) -> T_co:
-        if self._reset_iter:
-            raise StopIteration
-        try:
-            return next(self._datapipe_iter)  # type: ignore[arg-type]
-        except PauseIteration:
-            raise StopIteration
-        except StopIteration:
-            if self.reading_service is not None:
-                self.reading_service.finalize_iteration()
-            self._reset_iter = True
-            raise
```

Comment:
Since we are returning a wrapper in `__iter__`, I think it would be better for users to not call `__next__` directly on `DataLoader2`, but rather, call `next` on the `iterator` that created from `__iter__`. 

This should prevent the situation where `next` can be called on both `DataLoader2` and the `iterator` that has been created.

Thoughts?